src/TextField/TextFieldLabel.js

```diff
@@ -35,13 +43,30 @@ export default function TextFieldLabel(props, context) {
   }, classNameProp);
 
   return (
-    <label className={className} {...other} />
+    <label className={className} {...other}>
+      {children}
+      {required ? <span className={asteriskClassName}>{'\u2009'}*</span> : null}
```

Comment:
Still need to be cautious. If X in `X && B` isn't a boolean, we could have some surprise 😁 .